README.md

```diff
@@ -8,43 +8,105 @@ This repo builds a Docker image which collects tooling used to setup environment
 TTY_OPTIONS=$(shell [ -t 0 ] && echo '-it')
+TEMP_ENV_FILE:=$(shell mktemp)
+
 ifndef ENV
 $(error Need to set ENV)
 endif
 ifndef DIR
 $(error Need to set DIR)
 endif
 
-prepare:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) prepare $(DIR) $(ENV) $(SUFFIX)
-
-plan:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) plan $(DIR) $(ENV) $(SUFFIX) $(OPA_BLAST_RADIUS)
-
-apply:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) apply $(DIR) $(ENV) $(SUFFIX)
-
-destroy:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) destroy $(DIR) $(ENV) $(SUFFIX)
-
-state-remove:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) state-remove $(DIR) $(ENV) $(SUFFIX)
-
-validate:
-        docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" -v $${PWD}/$(DIR)/.terraform/$(ENV).env:/tmp/$(ENV).env -v $(AZURE_CONFIG_DIR):/work/.azure -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE) validate $(DIR) $(ENV) $(SUFFIX)
+AZURE_DIR_MOUNT:=$(shell echo "-v $(AZURE_CONFIG_DIR):/work/.azure")
+COMMAND:=$(shell echo "docker run --user $(shell id -u) $(TTY_OPTIONS) --entrypoint "/opt/terraform.sh" --env-file $(TEMP_ENV_FILE) $(AZURE_DIR_MOUNT) -v $${PWD}/$(DIR):/tmp/$(DIR) -v $${PWD}/global.tfvars:/tmp/global.tfvars $(IMAGE)")
+CLEANUP_COMMAND:=$(shell echo "$(MAKE) --no-print-directory teardown TEMP_ENV_FILE=$(TEMP_ENV_FILE)")
+
+.PHONY: setup
+.SILENT: setup
+setup:
+	set -e
+
+	mkdir -p $(AZURE_CONFIG_DIR)
+	export AZURE_CONFIG_DIR="$(AZURE_CONFIG_DIR)"
+
+	if [ "$${CI}" == "true" ]; then
+		echo ARM_CLIENT_ID=$${servicePrincipalId} >> $(TEMP_ENV_FILE)
+		echo ARM_CLIENT_SECRET=$${servicePrincipalKey} >> $(TEMP_ENV_FILE)
+		echo ARM_TENANT_ID=$${tenantId} >> $(TEMP_ENV_FILE)
+	fi
+
+	echo ARM_SUBSCRIPTION_ID=$$(az account show -o tsv --query 'id') >> $(TEMP_ENV_FILE)
+
+	if [ "$(AWS_ENABLED)" == "true" ]; then
+		if [ "$${CI}" == "true" ]; then
```

Comment:
I think I would have preferred `if [ -z "$AWS_ACCESS_KEY_ID" ] ; then <go get some credentials> ; fi` as that would allow both CI and users to provide creds via env vars.